kqueue.go

```diff
@@ -71,10 +71,8 @@ func (w *Watcher) Close() error {
 		return nil
 	}
 	w.isClosed = true
-	w.mu.Unlock()
 
 	// copy paths to remove while locked
-	w.mu.Lock()
 	var pathsToRemove = make([]string, 0, len(w.watches))
 	for name := range w.watches {
```

Comment:
I thought the same thing initially but `w.watches` is a map, so we can't `append()` directly, we need to iterate over its keys.